src/api/app/models/user.rb

```diff
@@ -99,104 +147,70 @@ def self.default_user_state
     self.login_failure_count = 0 if self.login_failure_count.nil?
   end
 
-  # Set the last login time etc. when the record is created at first.
-  def before_create
-    self.last_logged_in_at = Time.now
-  end
-
-  # Override the accessor for the "password_hash_type" property so it sets
-  # the "@new_hash_type" private property to signal that the password's
-  # hash method has been changed. Changing the password hash type is only
-  # possible if a new password has been provided.
-  def password_hash_type=(value)
-    write_attribute(:password_hash_type, value)
-    @new_hash_type = true
-  end
+  # After validation, the password should be encrypted
+  after_validation(:on => :create) do
+    if errors.empty? and @new_password and !password.nil?
+      # generate a new 10-char long hash only Base64 encoded so things are compatible
+      self.password_salt = [Array.new(10){rand(256).chr}.join].pack('m')[0..9]
 
-  # After saving, we want to set the "@new_hash_type" value set to false
-  # again.
-  after_save '@new_hash_type = false'
+      # vvvvvv added this to maintain the password list for lighttpd
```

Comment:
Duno. I just moved this up to the other callbacks. So the first 4 commits "Group foo" are just moving code, but not changing it.

@adrianschroeter Do you know if this is still valid / needed?
That line here ->
```
      # vvvvvv added this to maintain the password list for lighttpd
      write_attribute(:password_crypted, password.crypt('os'))
      #  ^^^^^^
```